Connection Issue:  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!

(Filling out the following with as much detail as you can provide will help us solve your issue sooner.)

### `@slack/bolt` version

4.2.1

### Your `App` and Receiver Configuration

const slackAppProvider = {
  provide: App,
  useFactory: (configService: ConfigService) => {
    const botToken = configService.get<string>('SLACK_BOT_TOKEN');
    const signingSecret = configService.get<string>('SLACK_SIGNING_SECRET');
    const appToken = configService.get<string>('SLACK_APP_LEVEL_TOKEN');

    if (!botToken || !signingSecret || !appToken) {
      throw new Error('Missing required Slack environment variables');
    }

    return new App({
      token: botToken,
      signingSecret: signingSecret,
      appToken: appToken,
      socketMode: true,
    });
  },
  inject: [ConfigService],
};

### Node.js runtime version

node:22-alpine

### Steps to reproduce:

(Share the commands to run, source code, and project settings)

1.
2.
3.

### Expected result:
Hello i get sometimes issues with the connection to the slack server which leads that my customers can use my app till it reconnects correctly.
It is often like  it has issue for 60 seconds then it reconnects agian correctly and it works then correctly till next connection Issue comes

### Actual result:

Here is what my app logs

[WARN]  socket-mode:SlackWebSocket:13 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:13 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:13 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:13 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:13 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:13 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A ping wasn't received from the server before the timeout of 30000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[WARN]  socket-mode:SlackWebSocket:14 A pong wasn't received from the server before the timeout of 5000ms!
[ERROR]  socket-mode:SocketModeClient:0 Failed to send a message as the client has no active connection
[NestWinston] 18 2025-04-11 11:35:59   ERROR  In DynamicIssueSelectedActionService in execute method method Ticket went something wrong Error: Failed to send a WebSocket message as the client is not ready - { stack: [ null ] }
[ERROR]  bolt-app UnknownError: In DynamicIssueSelectedActionService in execute method method Ticket went something wrong Error: Failed to send a WebSocket message as the client is not ready
    at asCodedError (/app/node_modules/@slack/bolt/dist/errors.js:47:12)
    at App.handleError (/app/node_modules/@slack/bolt/dist/App.js:691:59)
    at App.processEvent (/app/node_modules/@slack/bolt/dist/App.js:676:25)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async SocketModeClient.<anonymous> (/app/node_modules/@slack/bolt/dist/receivers/SocketModeReceiver.js:142:17) {
  code: 'slack_bolt_unknown_error',
  original: Error: In DynamicIssueSelectedActionService in execute method method Ticket went something wrong Error: Failed to send a WebSocket message as the client is not ready
      at DynamicIssueSelectedActionService.execute (/app/dist/slack/actions/dynamic-issue-selected-action.service.js:226:19)
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
      at async SlackActionController.handleDynamicChildIssueSelect (/app/dist/slack/actions/slack.action.controller.js:31:9)
      at async Array.<anonymous> (/app/node_modules/@slack/bolt/dist/middleware/builtin.js:161:9)
      at async Array.onlyActions (/app/node_modules/@slack/bolt/dist/middleware/builtin.js:30:9)
      at async Promise.allSettled (index 5)
      at async /app/node_modules/@slack/bolt/dist/App.js:662:48
      at async Array.<anonymous> (/app/node_modules/@slack/bolt/dist/conversation-store.js:67:9)
      at async Array.ignoreSelf (/app/node_modules/@slack/bolt/dist/middleware/builtin.js:264:5)
      at async App.processEvent (/app/node_modules/@slack/bolt/dist/App.js:640:13)
}
[ERROR]   An unhandled error occurred while Bolt processed (type: block_actions, error: Error: In DynamicIssueSelectedActionService in execute method method Ticket went something wrong Error: Failed to send a WebSocket message as the client is not ready)
[ERROR]  socket-mode:SocketModeClient:0 Failed to send a message as the client has no active connection

--- a/src/receivers/SocketModeReceiver.ts
+++ b/src/receivers/SocketModeReceiver.ts
@@ -39,6 +39,10 @@ export interface SocketModeReceiverOptions {
   installerOptions?: InstallerOptions;
   appToken: string; // App Level Token
   customRoutes?: CustomRoute[];
+  clientPingTimeout?: number;
+  serverPingTimeout?: number;
+  pingPongLoggingEnabled?: boolean;
+  autoReconnectEnabled?: boolean;
   // biome-ignore lint/suspicious/noExplicitAny: user-provided custom properties can be anything
   customPropertiesExtractor?: (args: any) => StringIndexed;
   processEventErrorHandler?: (args: SocketModeReceiverProcessEventErrorHandlerArgs) => Promise<boolean>;
@@ -96,6 +100,10 @@ export default class SocketModeReceiver implements Receiver {
     appToken,
     logger = undefined,
     logLevel = LogLevel.INFO,
+    clientPingTimeout = undefined,
+    serverPingTimeout = undefined,
+    pingPongLoggingEnabled = undefined,
+    autoReconnectEnabled = undefined,
     clientId = undefined,
     clientSecret = undefined,
     stateSecret = undefined,
@@ -111,6 +119,10 @@ export default class SocketModeReceiver implements Receiver {
       appToken,
       logLevel,
       logger,
+      clientPingTimeout,
+      serverPingTimeout,
+      pingPongLoggingEnabled,
+      autoReconnectEnabled,
       clientOptions: installerOptions.clientOptions,
     });
 

--- a/test/unit/receivers/SocketModeReceiver.spec.ts
+++ b/test/unit/receivers/SocketModeReceiver.spec.ts
@@ -1,3 +1,4 @@
+import { EventEmitter } from 'node:events';
 import { IncomingMessage, ServerResponse } from 'node:http';
 import path from 'node:path';
 import { InstallProvider } from '@slack/oauth';
@@ -95,6 +96,115 @@ describe('SocketModeReceiver', () => {
         customPropertiesExtractor: ({ type, body }) => ({ payload_type: type, body }),
       });
       assert.isNotNull(receiver);
+    });
+    it('should pass clientPingTimeout to SocketModeClient', async () => {
+      const constructorSpy = sinon.spy();
+      class FakeSocketModeClient extends EventEmitter {
+        // biome-ignore lint/suspicious/noExplicitAny: test mock
+        constructor(opts: any) {
+          super();
+          constructorSpy(opts);
+        }
+      }
+      const smOverrides = mergeOverrides(overrides, {
+        '@slack/socket-mode': {
+          SocketModeClient: FakeSocketModeClient,
+        },
+      });
+      const SocketModeReceiver = importSocketModeReceiver(smOverrides);
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        clientPingTimeout: 15000,
+      });
+      assert.isNotNull(receiver);
+      sinon.assert.calledOnce(constructorSpy);
+      const constructorArgs = constructorSpy.firstCall.args[0];
+      assert.equal(constructorArgs.clientPingTimeout, 15000);
+    });
+    it('should pass serverPingTimeout to SocketModeClient', async () => {
+      const constructorSpy = sinon.spy();
+      class FakeSocketModeClient extends EventEmitter {
+        // biome-ignore lint/suspicious/noExplicitAny: test mock
+        constructor(opts: any) {
+          super();
+          constructorSpy(opts);
+        }
+      }
+      const smOverrides = mergeOverrides(overrides, {
+        '@slack/socket-mode': {
+          SocketModeClient: FakeSocketModeClient,
+        },
+      });
+      const SocketModeReceiver = importSocketModeReceiver(smOverrides);
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        serverPingTimeout: 60000,
+      });
+      assert.isNotNull(receiver);
+      sinon.assert.calledOnce(constructorSpy);
+      const constructorArgs = constructorSpy.firstCall.args[0];
+      assert.equal(constructorArgs.serverPingTimeout, 60000);
+    });
+    it('should pass all socket mode timeout options to SocketModeClient', async () => {
+      const constructorSpy = sinon.spy();
+      class FakeSocketModeClient extends EventEmitter {
+        // biome-ignore lint/suspicious/noExplicitAny: test mock
+        constructor(opts: any) {
+          super();
+          constructorSpy(opts);
+        }
+      }
+      const smOverrides = mergeOverrides(overrides, {
+        '@slack/socket-mode': {
+          SocketModeClient: FakeSocketModeClient,
+        },
+      });
+      const SocketModeReceiver = importSocketModeReceiver(smOverrides);
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+        clientPingTimeout: 15000,
+        serverPingTimeout: 60000,
+        pingPongLoggingEnabled: true,
+        autoReconnectEnabled: false,
+      });
+      assert.isNotNull(receiver);
+      sinon.assert.calledOnce(constructorSpy);
+      const constructorArgs = constructorSpy.firstCall.args[0];
+      assert.equal(constructorArgs.clientPingTimeout, 15000);
+      assert.equal(constructorArgs.serverPingTimeout, 60000);
+      assert.equal(constructorArgs.pingPongLoggingEnabled, true);
+      assert.equal(constructorArgs.autoReconnectEnabled, false);
+    });
+    it('should use defaults when socket mode timeout options are not provided', async () => {
+      const constructorSpy = sinon.spy();
+      class FakeSocketModeClient extends EventEmitter {
+        // biome-ignore lint/suspicious/noExplicitAny: test mock
+        constructor(opts: any) {
+          super();
+          constructorSpy(opts);
+        }
+      }
+      const smOverrides = mergeOverrides(overrides, {
+        '@slack/socket-mode': {
+          SocketModeClient: FakeSocketModeClient,
+        },
+      });
+      const SocketModeReceiver = importSocketModeReceiver(smOverrides);
+      const receiver = new SocketModeReceiver({
+        appToken: 'my-secret',
+        logger: noopLogger,
+      });
+      assert.isNotNull(receiver);
+      sinon.assert.calledOnce(constructorSpy);
+      const constructorArgs = constructorSpy.firstCall.args[0];
+      assert.equal(constructorArgs.appToken, 'my-secret');
+      assert.isUndefined(constructorArgs.clientPingTimeout);
+      assert.isUndefined(constructorArgs.serverPingTimeout);
+      assert.isUndefined(constructorArgs.pingPongLoggingEnabled);
+      assert.isUndefined(constructorArgs.autoReconnectEnabled);
     });
     it('should throw an error if redirect uri options supplied invalid or incomplete', async () => {
       const SocketModeReceiver = importSocketModeReceiver(overrides);

